commands/audit/jasrunner.go

```diff
@@ -2,62 +2,119 @@ package audit
 	"github.com/jfrog/jfrog-cli-security/commands/audit/jas/sast"
 	"github.com/jfrog/jfrog-cli-security/commands/audit/jas/secrets"
 	"github.com/jfrog/jfrog-cli-security/utils"
-	"github.com/jfrog/jfrog-client-go/utils/io"
 	"github.com/jfrog/jfrog-client-go/utils/log"
 )
 
-func runJasScannersAndSetResults(scanResults *utils.Results, directDependencies []string,
-	serverDetails *config.ServerDetails, workingDirs []string, progress io.ProgressMgr, thirdPartyApplicabilityScan bool, msi string) (err error) {
+func RunJasScannersAndSetResults(auditParallelRunner *utils.AuditParallelRunner, scanResults *utils.Results, serverDetails *config.ServerDetails,
+	auditParams *AuditParams, jfrogAppsConfig *jfrogappsconfig.JFrogAppsConfig, msi string) (err error) {
 	if serverDetails == nil || len(serverDetails.Url) == 0 {
 		log.Warn("To include 'Advanced Security' scan as part of the audit output, please run the 'jf c add' command before running this command.")
 		return
 	}
-	scanner, err := jas.NewJasScanner(workingDirs, serverDetails)
+	scanner, err := jas.NewJasScanner(serverDetails, jfrogAppsConfig)
 	if err != nil {
 		return
 	}
+	// Set environments variables for analytics in analyzers manager.
+	callback := jas.SetAnalyticsMetricsDataForAnalyzerManager(msi, scanResults.GetScaScannedTechnologies())
 	defer func() {
+		auditParallelRunner.ScannersWg.Wait()
+		callback()
 		cleanup := scanner.ScannerDirCleanupFunc
 		err = errors.Join(err, cleanup())
 	}()
-	if progress != nil {
-		progress.SetHeadlineMsg("Running applicability scanning")
-	}
-	// Set environments variables for analytics in analyzers manager.
-	callback := jas.SetAnalyticsMetricsDataForAnalyzerManager(msi, scanResults.GetScaScannedTechnologies())
-	defer callback()
-	scanResults.ExtendedScanResults.ApplicabilityScanResults, err = applicability.RunApplicabilityScan(scanResults.GetScaScansXrayResults(), directDependencies, scanResults.GetScaScannedTechnologies(), scanner, thirdPartyApplicabilityScan)
-	if err != nil {
-		return
-	}
+
 	// Don't execute other scanners when scanning third party dependencies.
-	if thirdPartyApplicabilityScan {
-		return
+	if !auditParams.thirdPartyApplicabilityScan {
+		for _, module := range scanner.JFrogAppsConfig.Modules {
+			if !jas.ShouldSkipScanner(module, utils.Secrets) {
+				auditParallelRunner.ScannersWg.Add(1)
+				_, err = auditParallelRunner.Runner.AddTaskWithError(runSecretsScan(auditParallelRunner, scanner, scanResults, module), auditParallelRunner.AddErrorToChan)
+				if err != nil {
+					return fmt.Errorf("failed to create secrets scan task: %s", err.Error())
+				}
+			}
+			if !jas.ShouldSkipScanner(module, utils.IaC) {
+				auditParallelRunner.ScannersWg.Add(1)
+				_, err = auditParallelRunner.Runner.AddTaskWithError(runIacScan(auditParallelRunner, scanner, scanResults, module), auditParallelRunner.AddErrorToChan)
+				if err != nil {
+					return fmt.Errorf("failed to create iac scan task: %s", err.Error())
+				}
+			}
```

Comment:
```suggestion
        func addModuleJasScanTask(module jfrogappsconfig.Module, scanType utils.JasScanType, auditParallelRunner *utils.AuditParallelRunner, task parallel.TaskFunc) (err error) {
            if jas.ShouldSkipScanner(module, scanType) {
                return
            }
            auditParallelRunner.ScannersWg.Add(1)
            if _, err = auditParallelRunner.Runner.AddTaskWithError(); err != nil {
				err = fmt.Errorf("failed to create %s scan task: %s", scanType.toString() , err.Error())
			}
            return
        }
```
This part could also be extracted to a func to reduce duplications